### PR TITLE
tyr: only geojson polygon accepted

### PR DESCRIPTION
**Pull request by patochectp**

### CODE REVIEW

File: source/navitiacommon/navitiacommon/parser_args_type.py [hunk 1]
```diff
@@ -92,16 +92,12 @@ def is_geometry_valid(geometry):
...
-            features= value.get('features')
-            if not features:
-                is_valid_object = is_geometry_valid(value)
-            else:
-                for feature in features:
-                    geometry = feature.get('geometry')
-                    is_valid_object = is_valid_object and is_geometry_valid(geometry)
-            if not is_valid_object:
-                raise ValueError('invalid geometry')
+            type= value.get('type')
```
Comment (antoine-de) [inline source/navitiacommon/navitiacommon/parser_args_type.py]: spacing :smile:

File: source/navitiacommon/navitiacommon/parser_args_type.py [hunk 2]
```diff
@@ -92,16 +92,12 @@ def is_geometry_valid(geometry):
...
+            if not (type.lower() == 'polygon' if type else False):
```
Comment (antoine-de) [inline source/navitiacommon/navitiacommon/parser_args_type.py]: isn't it easier to write this:
`if not type or type.lower() != 'polygon':` ?

File: source/tyr/tests/integration/users_test.py [hunk 3]
```diff
@@ -209,77 +158,43 @@ def check(u):
...
     assert len(resp) == 1
     check(resp[0])
     assert resp[0]['shape'] == {}
     assert mock_rabbit.called
 
-def test_add_user_with_geojson(mock_rabbit, geojson):
+
+def test_add_user_with_multipolygon(mock_rabbit, geojson_multipolygon):
     """
     creation of a user passing arguments as a json
```
Comment (antoine-de) [inline source/tyr/tests/integration/users_test.py]: the comment is not up to date

File: source/tyr/tests/integration/users_test.py [hunk 4]
```diff
@@ -291,44 +206,49 @@ def test_add_user_with_invalid_geojson(mock_rabbit, invalid_geojsonfixture):
...
     resp = api_post('/v0/users/', data=user)
+
     def check(u):
-        for k,_ in user.iteritems():
+        for k, _ in user.iteritems():
             assert u[k] == user[k]
         assert u['end_point']['name'] == 'navitia.io'
         assert u['type'] == 'with_free_instances'
-        assert u['block_until'] == None
+        assert u['block_until'] is None
```
Comment (antoine-de) [inline source/tyr/tests/integration/users_test.py:244]: :+1:

File: source/tyr/tests/integration/users_test.py [hunk 5]
```diff
@@ -655,13 +589,16 @@ def test_get_users(create_multiple_users):
     """
     resp = api_get('/v0/users')
 
-    assert resp[0]['has_shape'] == True
-    assert resp[0]['shape'] == {}
-    assert resp[1]['has_shape'] == False
-    assert resp[1]['shape'] == None
+    for user in resp:
```
Comment (antoine-de) [inline source/tyr/tests/integration/users_test.py]: It's completly minor, but I prefer to write this:

```python
foo = next((u for u in resp if u.get('login') == 'foo'), None)
assert foo
assert foo.get('has_shape') is True
...

```
and same for the other user (you should get his name)

I think it's more robust and the test intent is clearer

File: source/navitiacommon/navitiacommon/parser_args_type.py [hunk 6]
```diff
@@ -92,16 +92,14 @@ def is_geometry_valid(geometry):
...
-            if not features:
-                is_valid_object = is_geometry_valid(value)
-            else:
-                for feature in features:
-                    geometry = feature.get('geometry')
-                    is_valid_object = is_valid_object and is_geometry_valid(geometry)
-            if not is_valid_object:
-                raise ValueError('invalid geometry')
+
+            type= value.get('type')
```
Comment (xlqian) [inline source/navitiacommon/navitiacommon/parser_args_type.py]: hmmm, "type" is a reserved word in python(it's a built-in function), we'd better avoid using it as variable name :)

File: source/tyr/tests/integration/users_test.py [hunk 7]
```diff
@@ -291,44 +207,49 @@ def test_add_user_with_invalid_geojson(mock_rabbit, invalid_geojsonfixture):
...
 def test_add_user_with_plus(mock_rabbit):
     """
     creation of a user with a "+" in the email
     """
     user = {'login': 'user1+test@example.com', 'email': 'user1+test@example.com'}
     resp = api_post('/v0/users/', data=json.dumps(user), content_type='application/json')
+
     def check(u):
-        for k,_ in user.iteritems():
+        for k, _ in user.iteritems():
```
Comment (xlqian) [inline source/tyr/tests/integration/users_test.py]: Minor: if you want check the perfect equality of two dicts, just use `cmp() == 0` 
https://docs.python.org/2/library/functions.html#cmp

**Review by antoine-de — APPROVED** [review]

File: source/tyr/tests/integration/users_test.py (hunk 7 above)
Comment (xlqian) [reply source/tyr/tests/integration/users_test.py]: I just learned that `cmp` is removed from python3 .... 😭

Comment (l-vincent-l) [reply source/tyr/tests/integration/users_test.py]: What about
```python
 for k in user.iterkeys():
```

Comment (xlqian) [reply source/tyr/tests/integration/users_test.py]: or `for k in user`,  that works too :)

**Review by xlqian — APPROVED** [review]

File: source/navitiacommon/navitiacommon/parser_args_type.py [hunk 8]
```diff
@@ -92,16 +92,14 @@ def is_geometry_valid(geometry):
         if value:
             is_valid_object = isinstance(value, dict)
             if not is_valid_object:
```
Comment (azime) [inline source/navitiacommon/navitiacommon/parser_args_type.py]: minor: 
````python
if not isinstance(value, dict):
````

**Review by antoine-de — APPROVED** [review]

**Review by pbougue — APPROVED** [review]

**Review by TeXitoi — APPROVED** [review]

**Comment by pbougue** [comment]
Jenkins, test this please

**Pull request merged**